lib/elixir/lib/module.ex

```diff
@@ -1985,9 +2000,29 @@ defmodule Module do
     :error, :badarg -> []
   end
 
-  defp assert_not_compiled!(fun, module) do
+  defp assert_not_compiled!(function_name_arity, module) do
     open?(module) ||
-      raise ArgumentError,
-            "could not call #{fun} with argument #{inspect(module)} because the module is already compiled"
+      raise ArgumentError, assert_not_compiled_message(function_name_arity, module)
+  end
+
+  defp assert_not_compiled_message({function_name, arity} = function_name_arity, module) do
+    mfa = "Module.#{function_name}/#{arity}"
+
+    "could not call #{mfa} with module argument #{inspect(module)} because the module is already compiled" <>
```

Comment:
I think the previous "argument" was more accurate than "module argument". Besides, it doesn't fit with `Module.eval_quoted` :)

Reply:
```suggestion
    "could not call #{mfa} because the module #{inspect(module)} is already compiled" <>
```

Reply:
@fertapric you are correct.
@josevalim that's one beautiful concise message.